api/swagger.yml

```diff
@@ -69,6 +69,17 @@ components:
       description: delimiter used to group common prefixes by
       schema:
         type: string
+    
+    IfNonMatch:
+      in: header
+      name: If-None-Match
+      description: |
+        Currently supports only "*" to allow uploading an object only if one doesn't exist yet.
+      example: "*"
+      required: false
+      schema:
+        type: string
+        pattern: '^\*$' # Currently, only "*" is supported
```

Comment:
Please avoid this: server limitations should be enforced on the server not on the client.  Suppose I update my server to a version that also supports ETag.  This is the only thing that prevents me from using an old client version.

I know the previous implementation also did this.  I now believe that it was also wrong.  But back then we didn't try hard enough to keep old clients compatible where possible.

Reply:
Will remove pattern and add validation on server